### PR TITLE
DOC-4295 added aggregate query examples

### PR DESCRIPTION
**Pull request by andy-stark-redis**
These examples don't yet have equivalents in the published docs for other languages but [this PR](https://github.com/redis/docs/pull/689) includes the Markdown for the tabs that will appear in [this page](https://redis.io/docs/latest/develop/interact/search-and-query/query/aggregation/).

Note that the last example just contains a comment saying `Not supported in NRedisStack`. The code didn't work when I tried it, which I believe is a result of [this open issue](https://github.com/StackExchange/NRediSearch/issues/11) with NRedisStack.

### CODE REVIEW

File: tests/Doc/QueryAggExample.cs [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        Assert.Equal(
+            "Key: bicycle:7, Price: 430, Discounted: 387",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+
+        test1Row = res1.GetRow(4);
+        Assert.Equal(
+            "Key: bicycle:8, Price: 1200, Discounted: 1080",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
```
Comment (atakavci) [inline tests/Doc/QueryAggExample.cs:208]: not sure if Redis has a contract to have them in this exact same order in response.

File: tests/Doc/QueryAggExample.cs [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+            $"Type: {res3Row["type"]}, Num total: {res3Row["num_total"]}"
+        );
+        // REMOVE_END
+
+
+        // STEP_START agg4
+
+        // Not supported in NRedisStack.
+
+        // STEP_END
```
Comment (atakavci) [inline tests/Doc/QueryAggExample.cs:292]: could you share the code fragment you tried?

**Review by atakavci — CHANGES_REQUESTED** [review]

File: tests/Doc/QueryAggExample.cs (hunk 2 above)
Comment (andy-stark-redis) [reply tests/Doc/QueryAggExample.cs]: @atakavci I've added it after `// STEP_START agg4` as a separate commit. It basically just doesn't include the list of bicycles that were supposed to come from the ToList reducer. The Row object doesn't even seem to have an element for the bicycles list.

Comment (atakavci) [reply tests/Doc/QueryAggExample.cs]: @andy-stark-redis you are right ? we have a bug with getting list of items with result. 
To share more context, return type does not provide a way to contain a list of items as of now _(it is a Dictionary<string, RedisValue>)_. We might need to introduce a breaking change to fix this. 
Would you mind opening an issue for this? i ll pick it soon.

Comment (andy-stark-redis) [reply tests/Doc/QueryAggExample.cs:315]: @atakavci I've added [this issue](https://github.com/redis/NRedisStack/issues/342). I've replaced the dodgy code with the original missing example message. I'll edit the issue to add a link to the source file once this PR is merged.

Comment (atakavci) [reply tests/Doc/QueryAggExample.cs:315]: great! thanks..

**Review by atakavci — APPROVED** [review]

**Pull request merged**